***Importing my File in Program...***

In [ ]:
import pandas as pd
df = pd.read_csv("SmartPhones.csv")
df

SmartPhone  Storage  Price
0   Xiomi Poco X3 Pro       32  15000
1   Xiomi Poco X3 Pro       64  19000
2   Xiomi Poco X3 Pro      128  20999
3   Xiomi Poco X3 Pro      256  25990
4       iQ 007 Legend       32  32990
5       iQ 007 Legend       64  36990
6       iQ 007 Legend      128  39990
7       iQ 007 Legend      256  43990
8              iQ 007       32  29990
9              iQ 007       64  31990
10             iQ 007      128  35990
11             iQ 007      256  39990

***Using Sklearn OneHotEncoder***

*To Covert SmartPhones name into Labels (numbers)*

In [ ]:

from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()

In [ ]:
dfle = df
dfle.SmartPhone = le.fit_transform(dfle.SmartPhone)
dfle

SmartPhone  Storage  Price
0            0       32  15000
1            0       64  19000
2            0      128  20999
3            0      256  25990
4            2       32  32990
5            2       64  36990
6            2      128  39990
7            2      256  43990
8            1       32  29990
9            1       64  31990
10           1      128  35990
11           1      256  39990

In [ ]:
x = dfle[['SmartPhone','Storage']].values
x

array([[  0,  32],
       [  0,  64],
       [  0, 128],
       [  0, 256],
       [  2,  32],
       [  2,  64],
       [  2, 128],
       [  2, 256],
       [  1,  32],
       [  1,  64],
       [  1, 128],
       [  1, 256]])

In [ ]:
y = dfle.Price.values
y

array([15000, 19000, 20999, 25990, 32990, 36990, 39990, 43990, 29990,
       31990, 35990, 39990])

***Using OneHotEncoder to create Dummy Variables for each SmartPhones***

*IMPORTING*

In [ ]:
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer
ct = ColumnTransformer([('SmartPhone', OneHotEncoder(), [0])], remainder = 'passthrough')

In [ ]:
x = ct.fit_transform(x)
x

array([[  1.,   0.,   0.,  32.],
       [  1.,   0.,   0.,  64.],
       [  1.,   0.,   0., 128.],
       [  1.,   0.,   0., 256.],
       [  0.,   0.,   1.,  32.],
       [  0.,   0.,   1.,  64.],
       [  0.,   0.,   1., 128.],
       [  0.,   0.,   1., 256.],
       [  0.,   1.,   0.,  32.],
       [  0.,   1.,   0.,  64.],
       [  0.,   1.,   0., 128.],
       [  0.,   1.,   0., 256.]])

In [ ]:
x = x[:,1:]
x

array([[  0.,   0.,  32.],
       [  0.,   0.,  64.],
       [  0.,   0., 128.],
       [  0.,   0., 256.],
       [  0.,   1.,  32.],
       [  0.,   1.,  64.],
       [  0.,   1., 128.],
       [  0.,   1., 256.],
       [  1.,   0.,  32.],
       [  1.,   0.,  64.],
       [  1.,   0., 128.],
       [  1.,   0., 256.]])

***Now importing Modules to train our Model***

In [ ]:

from sklearn.linear_model import LinearRegression
model = LinearRegression()

In [ ]:
model.fit(x,y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

***Now lets Predict***

***Price of iQ007 64GB Varient SmartPhone is 31990 rs***

In [ ]:
model.predict([[1,0,64]])

array([31990.14275362])

In [ ]:
model.predict([[0,0,32]])

array([16318.90289855])

In [ ]:
model.predict([[0,1,256]])

array([44561.08188406])

***Checking acuracy of our model***

In [ ]:
model.score(x,y)

0.9884693272584822

***98%***